src/main/java/uk/gov/pay/directdebit/mandate/services/MandateService.java

```diff
@@ -126,10 +121,7 @@ public TokenExchangeDetails getMandateFor(String token) {
                 .map(mandateStateUpdateService::tokenExchangedFor)
                 .orElseThrow(TokenNotFoundException::new);
 
-        String transactionExternalId = mandate.getType().equals(MandateType.ONE_OFF)
-                ? retrieveTransactionForOneOffMandate(mandate.getExternalId()).getExternalId()
-                : null;
-        return new TokenExchangeDetails(mandate, transactionExternalId);
+        return new TokenExchangeDetails(mandate, null);
```

Comment:
Can the transactionExternalId field be removed from TokenExchangeDetails?

Reply:
done